### PR TITLE
Expose HTTP method in unary handlers

### PR DESCRIPTION
**Pull request by akshayjshah**
Fixes #502 to expose the HTTP method in handlers.

### CODE REVIEW

File: protocol.go [hunk 1]
```diff
@@ -181,6 +181,13 @@ func (hc *errorTranslatingHandlerConnCloser) Close(err error) error {
 	return hc.fromWire(closeErr)
 }
 
+func (hc *errorTranslatingHandlerConnCloser) getHTTPMethod() string {
+	if methoder, ok := hc.handlerConnCloser.(interface{ getHTTPMethod() string }); ok {
+		return methoder.getHTTPMethod()
+	}
+	return http.MethodPost
+}
+
```
Comment (jhump) [inline protocol.go:190]: I think this is unused.

Comment (jhump) [reply protocol.go:190]: No, never mind, I see.

**Review by jhump — APPROVED** [review]
Well, shoot. I now see my error -- I had a test verifying `GET` only on the client side and failed to add something server side for this. Sorry about that.

Thanks for fixing!

**Review by jchadwick-buf — APPROVED** [review]
Great catch Akshay! Very sneaky issue.

**Pull request merged**